### DISCUSSION TITLE
internal aedes subscription retained messages

### DISCUSSION THREAD

**Discussion by friesendrywall** [moscajs/aedes#846] [Q&A] [answered]
It would appear the direct internal subscriptions to aedes would not publish retained messages.  Is there any type of workaround, or am I doing something wrong here?

for example, I'm running a broker, and bridging to a mqtt 5 broker I have built, and when I subscribe for the mqtt5 relay subscription, I don't get any of the retained messages.

**Comment by robertsLando** [comment, marked as answer]
@friesendrywall what you can do is to read retained from persistence like we do in [subscribe](https://github.com/moscajs/aedes/blob/main/lib/handlers/subscribe.js#LL230C4-L243C4) or use mqttjs client instead to subscribe to the broker

**Comment by friesendrywall** [comment]
Thank you.
